src/centos/scripts/start-marklogic.sh

```diff
@@ -193,46 +193,46 @@ function validate_tls_parameters {
 ################################################################
 function validate_cert {
     local cacertfile=$1
-    local request response
-    request=$(curl -s -S -L --cacert "${cacertfile}" --ssl "${ML_BOOTSTRAP_PROTOCOL}"://"${MARKLOGIC_BOOTSTRAP_HOST}":8001 --anyauth --user "${ML_ADMIN_USERNAME}":"${ML_ADMIN_PASSWORD}")
+    local response
+    curl -s -S -L --cacert "${cacertfile}" --ssl "${ML_BOOTSTRAP_PROTOCOL}"://"${MARKLOGIC_BOOTSTRAP_HOST}":8001 --anyauth --user "${ML_ADMIN_USERNAME}":"${ML_ADMIN_PASSWORD}"
     response=$?
     if [ $response -ne 0 ]; then
         error "MARKLOGIC_JOIN_CACERT_FILE is not valid, please check above error for details. Node shutting down." exit
     fi
 }
 
 ################################################################
-# retry_and_timeout(target_url, expected_response_code, additional_options, return_error)
-# The third argument is optional and can be used to pass additional options to curl.
-# Fourth argurment is optional, default is set to true, can be used when custom error handling is required,
-# if set to true means function will return error and exit if curl fails N_RETRY times
-# setting to false means function will return response code instead of failing and exiting.
+# curl_retry_validate(return_error, endpoint, expected_response_code, curl_options)
 # Retry a curl command until it returns the expected response
 # code or fails N_RETRY times.
 # Use RETRY_INTERVAL to tune the test length.
 # Validate that response code is the same as expected response
 # code or exit with an error.
 #
-#   $1 :  The target url to test against
-#   $2 :  The expected response code
-#   $3 :  Additional options to pass to curl
-#   $4 :  Option to return error or response code in case of error   
+#   $1 :  Option to return error or response code in case of error 
+#   $2 :  The target url to test against
+#   $3 :  The expected response code
+#   $4+:  Additional options to pass to curl
 ################################################################
 function curl_retry_validate {
-    local retry_count response_code request
-    local return_error="${4:-true}" curl_options=$3 endpoint=$1
+    local retry_count response_code
+    local return_error=$1; shift
+    local endpoint=$1; shift
+    local expected_response_code=$1; shift
+    local curl_options=("$@")
+
     for ((retry_count = 0; retry_count < N_RETRY; retry_count = retry_count + 1)); do
-        request="curl -m 30 -s -w '%{http_code}' $curl_options $endpoint"
-        response_code=$(eval "${request}")
-        if [[ ${response_code} -eq $2 ]]; then
+        response_code=$(curl -m 30 -s -w '%{http_code}' "${curl_options[@]}" "$endpoint")
```

Comment:
We should capture the output from curl (the easiest is to send to a temp file) in case there is an error so the error can be logged. More logging will help with troubleshooting. You can create a new bug/task for this if you don't want to address in this fix.